### PR TITLE
Backport the global styles variations changes from Gutenberg

### PR DESCRIPTION
**Pull request by youknowriad**
This PR backports the global styles variations endpoint changes from the Gutenberg plugin.

Trac ticket: https://core.trac.wordpress.org/ticket/55505

### CODE REVIEW

File: tests/phpunit/tests/rest-api/rest-global-styles-controller.php [hunk 1]
```diff
@@ -455,4 +460,41 @@ public function test_get_item_schema() {
 		$this->assertArrayHasKey( 'settings', $properties, 'Schema properties array does not have "settings" key' );
 		$this->assertArrayHasKey( 'title', $properties, 'Schema properties array does not have "title" key' );
 	}
+
+	
+	public function test_get_theme_items() {
+		wp_set_current_user( self::$admin_id );
+		switch_theme( 'theme1' );
```
Comment (youknowriad) [inline tests/phpunit/tests/rest-api/rest-global-styles-controller.php]: I'm going to need some help here :). I noticed that there's a "theme1" theme in php unit test fixtures, I tried using it here but failed (see unit test failure). I'm not really sure how we do switch to that particular theme in a php unit test in core.

Comment (gziolo) [reply tests/phpunit/tests/rest-api/rest-global-styles-controller.php]: It should work, see:

https://github.com/WordPress/wordpress-develop/blob/26c6b33b870a7630202b317a515d9d6e9183c258/tests/phpunit/tests/theme/wpThemeJsonResolver.php#L59-L63

Unless some set up work is missing like there:

https://github.com/WordPress/wordpress-develop/blob/26c6b33b870a7630202b317a515d9d6e9183c258/tests/phpunit/tests/theme/wpThemeJsonResolver.php#L15-L22

**Comment by gziolo** [comment]
@oandregal and @jorgefilipecosta, we would appreciate your help here.

File: tests/phpunit/tests/rest-api/rest-global-styles-controller.php (hunk 1 above)
Comment (oandregal) [reply tests/phpunit/tests/rest-api/rest-global-styles-controller.php]: I see that there are two themes living at `tests/phpunit/data/themedir1`: `block-theme` and `block-theme-child`. To use any of those we need to register the `themedir1` as a theme root (so tests look fore themes in that directory as well). That's what the the resolver tests do (the link Greg shared).

**Comment by youknowriad** [comment]
I think I managed to fix the tests, let's see what CI says.

**Comment by oandregal** [comment]
@youknowriad it looks like the only failing test is `WP_Test_REST_Schema_Initialization::test_expected_routes_in_schema`, which is missing one of the global styles routes.

**Comment by youknowriad** [comment]
I guess we need to regenerate that REST API fixtures file, I forgot how to do that. I can the entry manually though.

**Comment by oandregal** [comment]
This is working as expected for me.

This is what I've done to test:

Created a `styles` folder within `src/wp-content/themes/twentytwentytwo` with two files.

`bluish.json` whose content was:

```json
{
  "version": 2,
  "title": "Blue style",
  "styles": {
    "color": {
      "background": "blue",
      "text": "white"
    }
  },
  "settings": {
    "color": {
      "palette": [
        { "slug": "blue", "name": "Blue", "color": "blue" }
      ]
    }
  }
}
```

`redish.json` whose content was:

```json
{
  "version": 2,
  "title": "Red style",
  "styles": {
    "color": {
      "background": "red",
      "text": "white"
    }
  },
  "settings": {
    "color": {
      "palette": [
        { "slug": "red", "name": "Red", "color": "red" }
      ]
    }
  }
}
```

Then, I opened the site editor and, in the console, executed this command: `wp.apiFetch({path: '/wp/v2/global-styles/themes/twentytwentytwo/variations', method: 'GET'})`. Once the request is resolved I saw the content of the style variations. 

I've also tested with the existing TwentyTwentyTwo theme with no `styles` folder and also works fine (the endpoint returns 0 style variations).

**Comment by youknowriad** [comment]
@gziolo Should I go ahead and commit this one?

**Comment by gziolo** [comment]
> @gziolo Should I go ahead and commit this one?

That would be fantastic, thank you 🙇🏻

**Comment by youknowriad** [comment]
committed in https://core.trac.wordpress.org/changeset/53072

**Pull request closed without merging**